src/main.cc

```diff
@@ -329,15 +329,22 @@ void onExit(void)
     LogEventDeinit();
 }
 
-gboolean terminationSignalHandler(gpointer data) {
+gboolean terminationSignalHandler(gpointer pPendingSignal)
```

Comment:
Sorry, I am not understanding what you mean about fixing all the commits into one. If I do that, are you going to be able to see the changes between reviews?
I know that I must pull into master everything as a single commit (that can be done by clicking in Github in "Squash and merge"), but I didn't know that we should keep a single commit while reviewing.

Reply:
Ah no worries, we do not tend to use squash and merge for better or worse and add a merge commit on top of the original commits, we tend to rebase and force push to the PR branch upon review. Point about seeing changes between reviews is valid - assumption here is that we trust that the changes will be acted upon accordingly. Maybe we should request a second PR marked "v2" instead? This would be the case when submitting patches on a mailing list - you issue a completely new set of patches marked "v2", but that might be an overkill.
I suppose we work the way we work because often times a PR might touch more modules than 1 and the commits themselves need to be tidy. It is easier to look at a single commit that is a final version of what you wanted to achieve than going through history of how you got there.

Reply:
I find useful for large changes to see single commits, which gives the chance to see how small bits and pieces are added, e.g. adding a singe parameter to a change which adds many of them. I usually use the commit selector from Github for that.
![image](https://github.com/The-Good-Penguin/tgp-krill-kounter/assets/169656900/f4dcbca9-8667-4927-9e22-9008e80cae44)
Anyway, I am going to squash my two commits into one following the rules.
I am very grateful for your explanation, I would have missed that detail from the coding flow if you don't tell me. Thank you.